src/_components/Navbar.jsx

```diff
@@ -1,29 +1,109 @@
 import React from "https://esm.sh/react";
 
-export default function Navbar() {
+// Global navigation links array
+const navigationLinks = [
+  { href: "/mission", text: "Mission" },
+  { href: "/projects", text: "Projects" },
+  { href: "/community", text: "Get Involved" },
+  { href: "/blog", text: "Blog" },
+];
+
+export default function NavigationBar() {
   const logo = "/assets/logos/logo_negative.png";
   return (
-    <nav className="flex items-center justify-around space-x-64 p-6 bg-primary z-50">
-      <a href="/" className="flex items-center text-white mr-60">
-        <img src={logo} alt="Logo" className="mr-2 w-10 h-10" />
-        <span className="self-center text-3xl font-semibold font-poppins whitespace-nowrap dark:text-white">
+    <nav className="relative flex items-center justify-between p-6 bg-primary z-50">
+      {/* Logo and Brand Name Section */}
+      <a href="/" className="flex items-center text-white">
+        <img src={logo} alt="Blueprint Logo" className="mr-2 w-10 h-10" />
+        <span className="text-3xl font-poppins font-semibold hover:font-bold">
           blueprint
         </span>
       </a>
-      <div>
-        <a href="/mission" className="mr-4 text-xl text-white">
-          Mission
-        </a>
-        <a href="/projects" className="mr-4 text-xl text-white">
-          Projects
-        </a>
-        <a href="/community" className="mr-4 text-xl text-white">
-          Get Involved
-        </a>
-        <a href="/blog" className="text-xl text-white">
-          Blog
-        </a>
+      {/* Desktop Navigation Menu - Visible only on large screens */}
+      <div className="hidden lg:flex space-x-6">
+        {renderDesktopNavigationLinks()}
+      </div>
+      {/* Mobile Menu Toggle and Dropdown - Visible on screens smaller than "large" */}
+      <div className="lg:hidden relative">
+        {/* Toggles the mobile menu */}
+        <input
```

Comment:
Could you instead use a `useState` hook here. I don't think it is the best approach to use an input checkbox to toggle the menu with the links. You could maybe do something like
```
const [isMobileMenuOpen, setIsMobileMenuOpen] = useState(false);
{/* Code */}
        <button
          onClick={() => setIsMobileMenuOpen(!isMobileMenuOpen)}
          className="block cursor-pointer p-2"
          aria-label="Toggle mobile menu"
          aria-expanded={isMobileMenuOpen}
        >
          {renderHamburgerIcon()}
        </button>
```

Reply:
Using this checkbox approach is mainly concerning for accessibility reasons. Per our talk @ZhaoYuLL, I know i mentioned staying away from using the react state for the tracking window size changes as it was weird, but I think this true/false state toggle is a valid approach. My fault on the misdirection tho 

If anyone is interested, here are some discussions on this "checkbox hack" topic: [some dev article](https://justmarkup.com/articles/2019-12-04-hamburger-menu/#the-checkbox-hack) and [a related reddit post](https://www.reddit.com/r/webdev/comments/10gpdxn/the_checkbox_hack_pros_and_cons/).

Reply:
Didn't know about this. You can leave it as is.

Reply:
Ok I'll leave it as is for now